service/history/timerQueueProcessorBase.go

```diff
@@ -360,6 +360,12 @@ func (t *timerQueueProcessorBase) readAndFanoutTimerTasks() (*persistence.TimerT
 	}
 
 	if !moreTasks {
+		if lookAheadTask == nil { // try to get lookAheadTask
+			lookAheadTask, err = t.timerQueueAckMgr.readLookAheadTask()
```

Comment:
i believe you should combine this readLookAheadTask  function and readTimerTasks function.

Reply:
updated